Sources/NetworkProtection/KeyManagement/NetworkProtectionKeyStore.swift

```diff
@@ -103,6 +110,10 @@ public final class NetworkProtectionKeychainKeyStore: NetworkProtectionKeyStore
         return KeyPair(privateKey: currentPrivateKey, expirationDate: currentExpirationDate)
     }
 
+    public func newKeyPair() -> KeyPair {
+        return newCurrentKeyPair()
```

Comment:
This is incorrect as it'll update the key pair internally - instead we need to generate a new one but _not_store it anywhere until rekeying is done. I'll fix this.

Reply:
Fixed now

Reply:
Just curious - is this in-progress still?

Reply:
Nope, all done now - everything in all of these PRs is final, unless you find a bug. 😄